src/main/java/org/justinbaur/bankteller/service/AdminServiceDatabaseImpl.java

```diff
@@ -1,29 +1,85 @@
 package org.justinbaur.bankteller.service;
 
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.justinbaur.bankteller.domain.Account;
+import org.justinbaur.bankteller.domain.Profile;
+import org.justinbaur.bankteller.domain.Address;
+import org.justinbaur.bankteller.domain.CustomerInfo;
 import org.justinbaur.bankteller.exceptions.AccountNotFound;
-import org.justinbaur.bankteller.repository.AccountRepository;
+import org.justinbaur.bankteller.exceptions.ProfileNotFound;
+import org.justinbaur.bankteller.repository.ProfileRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AdminServiceDatabaseImpl extends ProfileServiceImpl implements AdminService {
```

Comment:
Can you explain the reasoning behind extending `ProfileServiceImpl`

Reply:
Both AdminService and UserProfileService are types of ProfileService. AdminServiceImpl makes extended use of the inherited method getProfile from ProfileService. Do you suggest I do this differently?